src/OrchardCore.Modules/OrchardCore.Users/Controllers/AdminController.cs

```diff
@@ -506,5 +506,42 @@ public async Task<IActionResult> EditPassword(ResetPasswordViewModel model)
 
             return View(model);
         }
+
+        [HttpPost]
+        public async Task<IActionResult> Unlock(string id)
+        {
+            if (!await _authorizationService.AuthorizeAsync(User, Permissions.ManageUsers))
+            {
+                return Forbid();
+            }
+
+            var user = await _userManager.FindByIdAsync(id) as User;
+
+            if (user == null)
+            {
+                return NotFound();
+            }
+
+            await _userManager.ResetAccessFailedCountAsync(user);
+            var result = await _userManager.SetLockoutEnabledAsync(user, false);
```

Comment:
This should not disable/enable lock out, but clear the `LockedOutUtc` of this user.

Reply:
Added `SetLockoutEndDateAsync(user, null);`, Kept `SetLockoutEnabledAsync`

Reply:
This permanently disables lock out for this user, while it should just unlock it.